addons/MountMuzzle/mountmuzzle.lua

```diff
@@ -102,84 +121,136 @@ function resolveCurrentMuzzle()
             colors.warn
         )
     end
+    
     return muzzles[current_muzzle]
 end
 
-function requestInject()
-    needs_inject = true
+function setMuzzle(muzzle)
+    settings.muzzle = muzzle
+    settings:save()
+end
+
+function playerInReive()
+    return getPlayerBuffs():contains(player.buffs.reiveMark)
+end
+
+function playerMounted()
+    return getPlayerBuffs()
+end
+
+function playerIsMounted()
+    local _player = windower.ffxi.get_player()
+    
+    if _player then
+        return _player.status == player.statuses.mounted or getPlayerBuffs():contains(player.buffs.mounted)
+    end
+    
+    return false 
+end
+
+function muzzleValid(muzzle)
+    return muzzles[muzzle] ~= nil
 end
 
 function injectMuzzleMusic()
+    injectMusic(music.types.mount, resolveCurrentMuzzle().song)
+end
+
+function injectMusic(bgmType, songID) 
     _packets.inject(_packets.new('incoming', packets.inbound.music_change.id, {
-        ['BGM Type'] = music.types.mount,
-        ['Song ID'] = resolveCurrentMuzzle().song,
+        ['BGM Type'] = bgmType,
+        ['Song ID'] = songID,
     }))
 end
 
+function requestInject()
+    needs_inject = true
+end
+
 function handleInjectionNeeds() 
     if needs_inject and playerIsMounted() then
         injectMuzzleMusic()
         needs_inject = false; 
     end
 end
 
-function playerIsMounted()
-    local _player = windower.ffxi.get_player()
-    if _player then
-        return mounted or _player.status == player.statuses.mounted
-    end
-    return false 
+function tryInject()
+    requestInject()
+    handleInjectionNeeds()
 end
 
-windower.register_event('login', requestInject)
-windower.register_event('load', requestInject)
-windower.register_event('zone change', requestInject)
+windower.register_event('login', function() 
+    tryInject()
+end)
+
+windower.register_event('load', function() 
+    tryInject()
+end)
+
+windower.register_event('unload', function() 
+    injectMusic(music.types.mount, muzzles.zone.song)
+end)
+
+windower.register_event('zone change', function() 
+    tryInject()
+end)
```

Comment:
Not too important, but you can chain events into a single `register_event` call:
```lua
windower.register_event('login', 'logout', 'zone change', function()
    tryInject()
end)
```

Reply:
oh, awesome, thanks!